examples/heat_pump/src/heat_pump_main.c

```diff
@@ -234,9 +248,32 @@ bool run_heat_pump_device(NabtoDevice* dev, struct heat_pump* heatPump, const st
             NabtoDeviceFuture* future = nabto_device_future_new(dev);
             nabto_device_device_events_init_listener(device, listener);
             NabtoDeviceEvent event;
+
+            int updatesPerSecond = 30;
+            double deltaTime = 1.0 / updatesPerSecond;
+            double currentTime = get_timestamp_seconds();
+            double accumulator = 0.0;
+
             while(true) {
                 nabto_device_listener_device_event(listener, future, &event);
-                ec = nabto_device_future_wait(future);
+                while (nabto_device_future_ready(future) == NABTO_DEVICE_EC_FUTURE_NOT_RESOLVED) {
+                    double newTime = get_timestamp_seconds();
+                    double elapsed = newTime - currentTime;
+                    currentTime = newTime;
+
+                    accumulator += elapsed;
+                    while (accumulator >= deltaTime) {
+                        heat_pump_update(heatPump, deltaTime);
+                        accumulator -= deltaTime;
+                    }
+
+                    // sleep until next update is needed
+                    if (elapsed < deltaTime) {
+                        sleepSeconds(deltaTime - elapsed);
```

Comment:
Could we use `nabto_device_future_timed_wait()` instead of `sleep() + nabto_device_future_ready()`?